napari/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,457 @@
+
+from typing import Union
+
+import numpy as np
+import scipy.signal as signal
+
+from .._base_layer import Layer
+from .._register import add_to_viewer
+from ..._vispy.scene.visuals import Line as LinesNode
+from ...util.event import Event
+from vispy.color import get_color_names
+
+from .view import QtVectorsLayer
+
+@add_to_viewer
+class Vectors(Layer):
+    """
+    Vectors layer renders lines onto the image.
+    There are currently two data modes supported:
+        1) image-like vectors where every pixel defines its own vector
+        2) coordinate-like vectors where x-y position is not fixed to a grid
+    Supports ONLY 2d vector field
+
+    Properties
+    ----------
+    vectors : np.ndarray
+        array of shape (N,4) or (N, M , 2)
+
+    data : np.ndarray
+        ABC implementation.
+        returns vectors property setter/getter
+
+    _original_data : np.ndarray
```

Comment:
do not include private properties anywhere but in the extended summary/notes sections